utils/filters_test.go

```diff
@@ -0,0 +1,561 @@
+			multiFilterKeys:  []string{"multi"},
+			inputFilters:     []string{"single=value", "multi=value"},
+		},
+		"invalid keys": {
+			singleFilterKeys: []string{"single"},
+			inputFilters:     []string{"single=value", "multi=value"},
+			wantError:        true,
+		},
+		"invalid filter input": {
+			singleFilterKeys: []string{""},
+			inputFilters:     []string{"invalid_input"},
+			wantError:        true,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run("AddFilters_"+name, func(t *testing.T) {
+			filters, err := NewFilters(test.singleFilterKeys, test.multiFilterKeys, []string{})
+			if err != nil {
+				t.Fatalf("Unexpected error when creating filters: '%s'", err.Error())
+			}
+
+			err = filters.AddFilters(test.inputFilters)
+			if test.wantError {
+				if err == nil {
+					t.Errorf("Expected error for AddFilters(%v), got nil", test.inputFilters)
+				}
+			} else if err != nil {
+				t.Errorf("Unexpected error for AddFilters(%v): '%s'", test.inputFilters, err.Error())
+			}
+		})
+	}
+}
+
+func TestFilters_AddFilter(t *testing.T) {
+	tests := map[string]struct {
+		singleFilterKeys []string
+		multiFilterKeys  []string
+		filterKey        string
+		filterValue      string
+		wantError        bool
+	}{
+		"single value": {
+			singleFilterKeys: []string{"single"},
+			filterKey:        "single", filterValue: "value",
+		},
+		"multi value": {
+			singleFilterKeys: []string{"single"}, multiFilterKeys: []string{"multi"},
+			filterKey: "multi", filterValue: "value",
+		},
+		"invalid key": {
+			multiFilterKeys: []string{"multi"}, filterKey: "single",
+			filterValue: "value", wantError: true,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run("AddFilter_"+name, func(t *testing.T) {
+			filters, err := NewFilters(test.singleFilterKeys, test.multiFilterKeys, []string{})
+			if err != nil {
+				t.Fatalf("Unexpected error when creating filters: '%s'", err.Error())
+			}
+
+			err = filters.AddFilter(test.filterKey + "=" + test.filterValue)
+			if test.wantError {
+				if err == nil {
+					t.Errorf(
+						"Expected error for AddFilter(%s=%s), got nil",
+						test.filterKey,
+						test.filterValue,
+					)
+				}
+			} else if err != nil {
+				t.Errorf("Unexpected error for AddFilter(%s=%s): '%s'", test.filterKey, test.filterValue, err.Error())
+			} else if slices.Contains(filters.SingleFilterKeys, test.filterKey) {
+				value, _ := filters.GetSingle(test.filterKey)
+				if value != test.filterValue {
+					t.Errorf("Expected filter value to be '%s', got '%s'", test.filterValue, value)
+				}
+			} else {
+				values, _ := filters.GetMulti(test.filterKey)
+				if !slices.Contains(values, test.filterValue) {
+					t.Errorf("Expected filter values to contain '%s', got %v", test.filterValue, values)
+				}
+			}
+		})
+	}
+
+	filters, _ := NewFilters([]string{"single"}, []string{}, []string{})
+	err := filters.AddFilter("invalid_filter")
+	if err == nil {
+		t.Errorf("Expected error for AddFilter(%s), got nil", "invalid_filter")
+	}
```

Comment:
What do you think about using `t.Run(...)` also for this test?

Reply:
Nice catch